src/pages/home/report/ContextMenu/ContextMenuActions.js

```diff
@@ -139,7 +139,7 @@ export default [
         successIcon: Expensicons.Checkmark,
         shouldShow: type => type === CONTEXT_MENU_TYPES.REPORT_ACTION,
         onPress: (closePopover, {reportAction, reportID}) => {
-            Report.updateLastReadActionID(reportID, reportAction.sequenceNumber, true);
```

Comment:
Good catch I totally missed that, I will add that in

Reply:
For this particular change, I think we only need to set the `lastReadSequenceNumbers[reportID] = lastReadSequenceNumber; ` because there should always be a sequenceNumber when we're marking as unread. And even if the report is loading we can set it.

Same for the others, I think we only need to set the `lastReadSequenceNumbers` b/c the report shouldn't be loading since we'll be updating with `OpenReport` if the report is loading. And the sequenceNumber should be there for the others.

Reply:
Ok that sounds right to me as the `unreadActionCount` will be updated (as well as it can be) when we set the `optimisticData`, but we maybe still need to update the `lastReadSequenceNumbers` map with the new value. 

All this code is kind of tricky because the "set local last read" method is more of a "set local last read and reset unread count" (which is totally confusing). I'm drafting something up to get more ideas about how to handle this long term.

Reply:
Honestly, I kinda want to kill the entire `lastReadSequenceNumbers` entirely if you're down for that 😄 

I started drafting some ideas here that make sense to me, but haven't tested them yet

https://github.com/Expensify/App/pull/9755

Reply:
Ok so going to set it here, and let you kill it in your PR